Kernel/Devices/KCOVDevice.cpp

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright (c) 2021, Patrick Meyer <git@the-space.agency>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/Assertions.h>
+#include <AK/NonnullOwnPtr.h>
+#include <Kernel/Devices/KCOVDevice.h>
+#include <Kernel/Devices/KCOVInstance.h>
+#include <Kernel/FileSystem/FileDescription.h>
+#include <LibC/sys/ioctl_numbers.h>
+
+#include <Kernel/Panic.h>
+
+namespace Kernel {
+
+HashMap<ProcessID, KCOVInstance*>* KCOVDevice::proc_instance;
+HashMap<ThreadID, KCOVInstance*>* KCOVDevice::thread_instance;
+
+UNMAP_AFTER_INIT NonnullRefPtr<KCOVDevice> KCOVDevice::must_create()
+{
+    return adopt_ref_if_nonnull(new KCOVDevice).release_nonnull();
```

Comment:
The non-nothrow operator new won't ever return null.
```suggestion
    return adopt_ref(new KCOVDevice);
```

Reply:
ah right, I think I didn't fix this when all the must_create()s were changed. I think it needs to be `return adopt_ref(*new KCOVDevice);` though